src/main/business/applicationFacade.js

```diff
@@ -336,6 +338,27 @@ class ApplicationFacade{
         return this.userController.updateUser(data);
     }
 
+    async generateTask(data) {
+        validateParameters(data, {taskType: PRIMITIVE_TYPES.STRING});
+        switch (data.taskType) {
+            case GENERATED_TASK_TYPES.TAG_ANSWER:
+                return this.#generateTagAnswerTask(data)
+            default:
+                throw new EMSError(TASK_PROCESS_ERROR_MSGS.INVALID_TASK_TYPE(data.taskType), TASK_PROCESS_ERROR_CODES.INVALID_TASK_TYPE);
+        }
+    }
+
+    async completeGeneratedTask(data) {
+        validateParameters(data, {taskType: PRIMITIVE_TYPES.STRING});
```

Comment:
Seemed missing...
What with an id to the task, how do you know what task has been done ?

Reply:
Nope, this is it. 
There is no task ID because it was generated on the fly. These are tasks of constant work done by the staff, it can always be done again and again and there's no need to mark a specific, in system task, as complete. All I need is the task results based on the task type.